drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -47,6 +65,114 @@ static unsigned int siwx91x_on_join(sl_wifi_event_t event,
 	return 0;
 }
 
+static int siwx91x_ap_enable(const struct device *dev, struct wifi_connect_req_params *params)
+{
+	struct siwx91x_dev *sidev = dev->data;
+	int ret;
+
+	sl_wifi_ap_configuration_t configuration = {
+		.encryption = SL_WIFI_DEFAULT_ENCRYPTION,
+		.credential_id = SL_NET_DEFAULT_WIFI_AP_CREDENTIAL_ID,
+		.rate_protocol = SL_WIFI_RATE_PROTOCOL_AUTO,
+		.options = 0,
+		.keepalive_type = SL_SI91X_AP_NULL_BASED_KEEP_ALIVE,
+		.beacon_interval = 100,
+		.client_idle_timeout = 0xFF,
+		.dtim_beacon_count = 3,
+		.maximum_clients = 4,
+		.beacon_stop = 0,
+		.tdi_flags = SL_WIFI_TDI_NONE,
+		.is_11n_enabled = 1,
+		.ssid.length = params->ssid_length,
+	};
+
+	if (params->band != WIFI_FREQ_BAND_UNKNOWN && params->band != WIFI_FREQ_BAND_2_4_GHZ) {
+		return -ENOTSUP;
+	}
+
+	if (params->channel == WIFI_CHANNEL_ANY) {
+		configuration.channel.channel = SL_WIFI_AUTO_CHANNEL;
+	} else {
+		configuration.channel.channel = params->channel;
+	}
+
+	if (siwx91x_bandwidth(params->bandwidth) < 0) {
+		return -EINVAL;
+	}
+
+	configuration.channel.bandwidth = siwx91x_bandwidth(params->bandwidth);
+	strncpy(configuration.ssid.value, params->ssid, params->ssid_length);
+
+	switch (params->security) {
+	case WIFI_SECURITY_TYPE_NONE:
+		configuration.security = SL_WIFI_OPEN;
+		sl_net_set_credential(configuration.credential_id, default_wifi_ap_credential.type,
+				      (const void *)default_wifi_ap_credential.data,
+				      default_wifi_ap_credential.data_length);
+		break;
+
+	case WIFI_SECURITY_TYPE_WPA_PSK:
+		configuration.security = SL_WIFI_WPA;
+		break;
+
+	case WIFI_SECURITY_TYPE_PSK:
+		configuration.security = SL_WIFI_WPA2;
+		break;
+
+	case WIFI_SECURITY_TYPE_PSK_SHA256:
+		configuration.security = SL_WIFI_WPA2;
+		if (params->mfp != WIFI_MFP_REQUIRED) {
+			LOG_ERR("MFP required");
+			return -EINVAL;
+		}
+
+		sl_si91x_set_join_configuration(SL_WIFI_AP_INTERFACE,
+						SL_SI91X_JOIN_FEAT_MFP_CAPABLE_REQUIRED);
```

Comment:
I think MFP is orthogonal to the authentication scheme. Probably this configuration should done outside of this switch case.

BTW, WPA2-PSK w/o MFP seems legit. Is it a limitation of Wiseconnect?

I also assume `configuration.security = SL_WIFI_WPA3` automatically enables MFP? If ye, we could check `params->mfp == WIFI_MFP_REQUIRED`.